### ISSUE TITLE
Padrino.root wrong value in boot.rb

### ISSUE THREAD

**Issue by kakubei** [padrino/padrino-framework#1614]
Hello, just after upgrading to 0.12, I'm having an issue where `Padrino.root` outputs to the wrong path in `boot.rb`. 

On the console, the path is correct, but in the boot.rb file (when Sphinx is trying to load for example), `Padrino.root` outputs to the root of the `Bundler` gem:

    /Users/<username>/.rbenv/versions/2.1.0/lib/ruby/gems/2.1.0/gems/bundler-1.5.2/lib/bundler 

This is extremely strange and didn't happen before. How can I make sure this gets set correctly?

What is even stranger is that I have this line at the top of boot.rb:

    PADRINO_ROOT = File.expand_path('../..', __FILE__) unless defined?(PADRINO_ROOT)

If I remove the `unless defined?` bit, both PADRINO_ROOT and `Padrino.root` are set correctly though I do get a warning of the constant already being initialized. I have no idea how it's getting set to the root of Bundler and why this never happened until the 0.12 update. 

Any ideas?

Thanks.

**Comment by krukgit** [comment]
I have the same issue with Padrino 0.12.3

```
puts Padrino.root('log')

App 8453 stdout: /usr/local/rvm/gems/ruby-2.1.5@global/gems/bundler-1.8.4/lib/bundler
```

**Comment by namusyaka** [comment]
@krukgit Could you provide the failing project with Gemfile.lock ?

**Comment by krukgit** [comment]
@namusyaka 
It's a big project. I'll try to replicate the error in a smaller project and attach it here later.

**Comment by krukgit** [comment]
I investigated the issue and it seems to be caused by `Bundler.require(:default)` added to `config.ru` by one of my developers. 

It only causes problems when running from passenger, WEBrick sets the constant correctly.

Since it doesn't seem to be a problem with padrino you can close the issue but in case you want to take a look and figure out why it only fails in passenger, feel free to check out the test case I prepared:

https://github.com/krukgit/padrino_test

Steps to reproduce:
1. Clone the repository
2. Run `bundle install`
3. Run `bundle exec passenger start`
4. Look for a line in logs starting with PADRINO_ROOT

**Comment by ujifgc** [comment]
@krukgit thank you for your sample app. The case was marginal but a bug nevertheless.

**Comment by nesquena** [comment]
Nice, thanks for patching this @ujifgc. Glad you are getting in a few additional fixes in time for 0.13.0
